src/iov.c

```diff
@@ -79,8 +79,13 @@ void ofi_consume_iov_desc(struct iovec *iov, void **desc,
 			sizeof(*desc) * (*iov_count - i));
 	*iov_count -= i;
 out:
-	iov[0].iov_base = (uint8_t *)iov[0].iov_base + to_consume;
-	iov[0].iov_len -= to_consume;
+	if (to_consume < iov[0].iov_len) {
+		iov[0].iov_base = (uint8_t *)iov[0].iov_base + to_consume;
+		iov[0].iov_len -= to_consume;
+	} else if (*iov_count) {
+		*iov_count--;
+		iov[0].iov_len = 0;
+	}
```

Comment:
What if `to_consume == 0 && *iov_count == 1`, does this branch still apply?

Same in ofi_consume_rma_iov

Reply:
In the issue I filed Sean seems to think this function should assume:
```
assert(to_consume && *iov_count);
assert(to_consume <= ofi_total_iov_len(iov, *iov_count));
```
I think I've modified to handle the *iov_count=0 case, so now I think we can assume:
```
assert(to_consume);
assert(to_consume <= ofi_total_iov_len(iov, *iov_count));
```
But reading this I think there are two cases to consider: more than 1 iov, with the first one being zero and the rest non-zero, and 1 remaining iov with 0.

For the first case, the loop will actually consume the zero-sized iov, even if we are consuming 0 bytes.  so I think that's fine.
For the second case, iov_count-- will consume the last iov.

Reply:
See conversation below with Sean.  We are asserting to_consume != 0 now.